src/Microsoft.PowerShell.Commands.Utility/commands/utility/convert-HTML.cs

```diff
@@ -364,9 +415,49 @@ protected override void BeginProcessing()
 
             if (!_fragment)
             {
-                WriteObject("<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//EN\"  \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\">");
+                if(!Transitional)
+                {
+                    WriteObject("<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Strict//EN\"  \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd\">");
+                }
+                else
+                {
+                    WriteObject("<!DOCTYPE html PUBLIC \"-//W3C//DTD XHTML 1.0 Transitional//EN\"  \"http://www.w3.org/TR/xhtml1/DTD/xhtml1-transitional.dtd\">");
+                }
                 WriteObject("<html xmlns=\"http://www.w3.org/1999/xhtml\">");
                 WriteObject("<head>");
+                if(_charsetSpecified)
+                {
+                    WriteObject("<meta charset=\"" + _charset + "\">");
+                }
+                if(_metaSpecified)
+                {
+                    List<string> useditems = new List<string>();
+                    foreach(string s in _meta.Keys)
+                    {
+                        if(!useditems.Contains(s)){
+                            switch(s.ToLower()){
+                                case "content-type":
+                                case "default-style":
+                                case "x-ua-compatible":
+                                    WriteObject("<meta http-equiv=\"" + s + "\" content=\"" + _meta[s] + "\">");
+                                    break;
+                                case "application-name":
+                                case "author":
+                                case "description":
+                                case "generator":
+                                case "keywords":
+                                case "viewport":
+                                    WriteObject("<meta name=\"" + s + "\" content=\"" + _meta[s] + "\">");
+                                    break;
+                                default:
+                                    Exception exc = new NotSupportedException(StringUtil.Format(ConvertHTMLStrings.MetaPropertyNotFound, s));
+                                    ThrowTerminatingError (new ErrorRecord(exc, "MetaPropertyNotFound", ErrorCategory.ParserError, null));
+                                    break;
+                            }
+                            useditems.Add(s);
+                        }
```

Comment:
seems like we should give a warning if we are dropping duplicates

Reply:
I have changed it to a warning and decided to keep the meta tags that don't match the switch.